### DISCUSSION TITLE
__layout.svelte is ugly... So is %sveltekit.head%

### DISCUSSION THREAD

**Discussion by MrAmericanMike** [sveltejs/kit#5816] [Ideas]
As Rich would say, we are a compiler, we can do whatever we want.

So following the line of `__layout.svelte is ugly` so are `%sveltekit.head%` and `%sveltekit.body%`

Why not change them to something like `<sveltekit:head />` and `<sveltekit:body />` or anything that makes them look less out of place.

**Comment by richarddavenport** [comment]
I could be wrong, but I think svelte uses the same syntax. _If_ this changes it would probably have a ripple effect.

**Reply by MrAmericanMike** [reply]
Well Svelte doesn't have control over `<head>` and for the body you can define it however you want, like `<div id="app">` as in main you tell Svelte where to look for. So in that case, the user is in control.

```js
const app = new App({
	target: document.querySelector("#app")
});
```

**Comment by alumpe** [comment]
Honestly, `%sveltekit.head%` looking out of place is a good thing, it makes it stand out in the editor, it screams "THE HEAD IS INSERTED HERE",  `<sveltekit:head />` would look just like any other html element. Also `<sveltekit:head />` already has a very specific purpose in svelte itself, it shouldn't be used as the point of insertion as well.

**Reply by MrAmericanMike** [reply]
> Also `<sveltekit:head />` already has a very specific purpose in svelte itself, it shouldn't be used as the point of insertion as well.

Svelte has `<sveltekit:head />` ? I either missed that or it's probably something different.

**Reply by emmnull** [reply]
Yes, [svelte does already have a `<svelte:head />` ](https://svelte.dev/docs#template-syntax-svelte-head)element that makes it possible to dynamically manage the contents of `document.head` from within routes/components.

**Reply by MrAmericanMike** [reply]
> Yes, [svelte does already have a `<svelte:head />` ](https://svelte.dev/docs#template-syntax-svelte-head)element that makes it possible to dynamically manage the contents of `document.head` from within routes/components.

Can you share link to where that is documented, because now you mention routes and that sounds like SvelteKit as Svelte doesn't have routes neither. (Unless I'm missing something)

**Reply by emmnull** [reply]
Sure thing: https://svelte.dev/docs#template-syntax-svelte-head (same link as attached to my previous reply). I mentioned routes thinking of SvelteKit, you are right on that. But this special element is from Svelte (SvelteKit routes are essentially components with extended functionalities for ssr and other kit features)

**Comment by Mlocik97** [comment]
`__layout.svelte` is about to change, and `%sveltekit.head%` is somewhat "weird" but better than your ideas, because they are not tags! They are placeholders!

Also I would most likely to get rid of whole `app.html`.

**Reply by mavdotj** [reply]
> Also I would most likely to get rid of whole `app.html`.

Why get rid of the whole app.html? I find it useful to add extra javascript and css links outside of the layout file.

**Reply by cyaris** [reply]
How can I get rid of it.

**Comment by chientrm** [comment]
> As Rich would say, we are a compiler, we can do whatever we want.
> 
> So following the line of `__layout.svelte is ugly` so are `%sveltekit.head%` and `%sveltekit.body%`
> 
> Why not change them to something like `<sveltekit:head />` and `<sveltekit:body />` or anything that makes them look less out of place.

They are just markups help the SSR server know where to place their rendering result. They're not parsed into XML tree so using a tag for either head or body is not useful. Also tag variant would confuse the rendering server. Ex, `<sveltekit:head />`, `<sveltekit:head/>`, `<sveltekit:head      />`

**Reply by ljoukov** [reply]
I personally find it somewhat confusing that app.html is *not* parsed as HTML and is thus not preprocessed and minified during SSR, so maybe this is worth thinking through more.

**Reply by chientrm** [reply]
> I personally find it somewhat confusing that app.html is _not_ parsed as HTML and is thus not preprocessed and minified during SSR, so maybe this is worth thinking through more.

It's actually been preprocess by ViteJS during both SSR and CSR including static assets handling. The markups `%sveltekit.head%` or `%sveltekit.body%` are retained. Then ViteJS return the result as a raw string to SvelteKit to handle the rest 🐸 .
